src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp

```diff
@@ -1561,7 +1561,7 @@ CompressDictMatMulf32::CompressDictMatMulf32(Context::Ref ctx) {
 }
 
 SliceLastMatmul::SliceLastMatmul() {
-    auto matmul = opp::wrap_type<ov::op::v0::MatMul>({opp::any_input(), opp::any_input()});
+    auto matmul = opp::wrap_type<ov::op::v0::MatMul, ov::op::v7::Einsum>({opp::any_input(), opp::any_input()});
```

Comment:
We take the right most value from the second dimension which could be incorrect for some cases